### ISSUE TITLE
Login & processing issues

### ISSUE THREAD

**Issue by chauffer** [asciinema/asciinema-server#295]
Hey there, I'm trying to setup asciinema, I copied most of the docker-compose file to set this up on Rancher[0]. 

I'm encountering the following issues:
- `/login/new` redirects me to `/not-gonna-happen`, showing `The page you were looking for doesn't exist.`
- `/a/1`, the demo recording, says `This recording is being pre-processed at the moment. It will open automatically in a few seconds.` even after hours of uptime
- `/docs` is also a 404
I have run `setup` initially, although the webserver was already running at that time.

[0]
```
version: '2'
volumes:
  aciinema-redis-data:
    external: true
    driver: rancher-nfs
  asciinema-postgres-data:
    external: true
    driver: rancher-nfs
  asciinema-asciinema-volumes-cache:
    external: true
    driver: rancher-nfs
  asciinema-asciinema-uploads:
    external: true
    driver: rancher-nfs
  asciinema-asciinema-tmp:
    external: true
    driver: rancher-nfs
  asciinema-asciinema-log:
    external: true
    driver: rancher-nfs
services:
  asciinema:
    image: asciinema/asciinema-server:develop
    environment:
      DATABASE_URL: ...
      REDIS_URL: ...
      SMTP_FROM_ADDRESS: ...
      SECRET_KEY_BASE: ...
      URL_SCHEME: https
      URL_HOST: ...
    stdin_open: true
    volumes:
    - asciinema-asciinema-uploads:/uploads
    - asciinema-asciinema-log:/app/log
    - asciinema-asciinema-tmp:/tmp
    - asciinema-asciinema-volumes-cache:/cache
    tty: true
    labels:
      io.rancher.container.pull_image: always
  redis:
    image: redis:alpine
    stdin_open: true
    volumes:
    - aciinema-redis-data:/data
    tty: true
    labels:
      io.rancher.container.pull_image: always
  postgres:
    image: postgres:alpine
    stdin_open: true
    volumes:
    - asciinema-postgres-data:/var/lib/postgresql/data
    tty: true
    labels:
      io.rancher.container.pull_image: always
```

**Comment by underyx** [comment]
Your first volume is misspelled as `aciinema-redis-data`, FYI.

**Comment by sickill** [comment]
You didn't export any ports here so I assume you use Rancher's load balancer to route to `asciinema` service. If so, then make sure you're pointing it to service's port 80, where nginx is running, because this internal nginx instance is routing proper paths to Rails and Elixir apps (they both run in the container).

**Comment by sickill** [comment]
And make sure you don't override container's default `CMD` which starts supervisord (which starts nginx, rails, phoenix).

**Comment by chauffer** [comment]
Thanks, I was forwarding to 3000 instead of 80 - my bad. 😅 

The login page now works, however the "This recording is being pre-processed at the moment. It will open automatically in a few seconds." message remains.

**Comment by chauffer** [comment]
@sickill Ideas?

**Comment by sickill** [comment]
@chauffer I think that's because you had routing wrong when you uploaded and it's been handled by the old Rails endpoint instead of the new one. Does this also happen for new uploads?

Btw, this also suggests that you'r using very old asciinema recorder version (0.9.8 maybe?)
I suggest upgrading to the latest one.

**Comment by joshisa** [comment]
I've been playing with asciinema + k8s for a 4-5 months now.  On a local k8s with no loadbalancer, I've had zero problems with getting the asciinema-player to load on the page.   However, when I use hosted k8s clusters with a loadbalancer in front, I've been unable to get the asciinema-player object to instantiate.  The rest of the app works flawlessly: recording uploads, login, emails, etc ... No console errors, no errors on the backend that I can see.  The only symptom is the ".... being pre-processed" message.  After some comparisons between my working page and the faulty page, I've narrowed it down to a failure within the asciinema-player.js file during the createdCallback call. 
 Here's an excerpt of the player javascript section

```
r.createdCallback = function() {
    var r = this,
        u = n(t(this, "cols", "width", 0, parseInt), t(this, "rows", "height", 0, parseInt), t(this, "autoplay", "autoPlay", !0, Boolean), t(this, "preload", "preload", !0, Boolean), t(this, "loop", "loop", !0, Boolean), t(this, "start-at", "startAt", 0, parseInt), t(this, "speed", "speed", 1, parseFloat), t(this, "idle-time-limit", "idleTimeLimit", null, parseFloat), t(this, "poster", "poster", null, e), t(this, "font-size", "fontSize"), t(this, "theme", "theme"), t(this, "title", "title"), t(this, "author", "author"), t(this, "author-url", "authorURL"), t(this, "author-img-url", "authorImgURL"), {
            onCanPlay: function() {
                r.dispatchEvent(new CustomEvent("loadedmetadata")), r.dispatchEvent(new CustomEvent("loadeddata")), r.dispatchEvent(new CustomEvent("canplay")), r.dispatchEvent(new CustomEvent("canplaythrough"))
            },
            onPlay: function() {
                r.dispatchEvent(new CustomEvent("play"))
            },
            onPause: function() {
                r.dispatchEvent(new CustomEvent("pause"))
            }
        });
    this.player = asciinema.player.js.CreatePlayer(this, this.getAttribute("src"), u)
}
```
For my working page, this seems to run fine and move forward in the script execution.  However, on my page behind the loadbalancer, this call seems to end up failing silently somewhere ... and ultimately results in no player being rendered on the home page.  Hope this helps hint where the issue may lie.

**Comment by sickill** [comment]
@joshisa when you run it on k8s, do you run 1 or more asciinema-server containers? Do you use S3 bucket for the files, or you use default local (filesystem) storage?

**Comment by joshisa** [comment]
@sickill I actually figured out where things were failing from a configuration perspective (not sure why the callback symptom occurs).  In my case, i discovered a configuration difference beyond the ingress where I was separating my mountpaths onto different persisted volume paths for the rails and elixir apps in separate k8s deployment yamls.  The apps started fine, but the player loading on the main page failed to load.  When I changed my deployment to a more tightly coupled pod set and reused the same mount point, the player was happy again.   So the ingress observation (delta) was not a factor at all in the symptom/behavior.
